webauthn4j-metadata/src/test/java/com/webauthn4j/metadata/JsonFileMetadataItemsProviderTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.webauthn4j.metadata;
+
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.webauthn4j.converter.util.JsonConverter;
+import com.webauthn4j.metadata.data.MetadataItem;
+import com.webauthn4j.response.attestation.authenticator.AAGUID;
+
+class JsonFileMetadataItemsProviderTest {
+
+    private JsonConverter jsonConverter = new JsonConverter();
+
+    @Test
+    void fetchMetadata() throws Exception {
+       List<Path> paths= new ArrayList<>(4);
+       paths.add(Paths.get(ClassLoader.getSystemResource("com/webauthn4j/metadata/JsonMetadataItem_fido2.json").toURI()));
+       paths.add(Paths.get(ClassLoader.getSystemResource("com/webauthn4j/metadata/JsonMetadataItem_u2f.json").toURI()));
+       paths.add(Paths.get(ClassLoader.getSystemResource("com/webauthn4j/metadata/JsonMetadataItem_uaf.json").toURI()));
+       paths.add(Paths.get(ClassLoader.getSystemResource("com/webauthn4j/metadata/JsonMetadataItem_unknown_protocol.json").toURI()));
+       JsonFileMetadataItemsProvider provider = new JsonFileMetadataItemsProvider(jsonConverter, paths);
+       readMetadataItem(provider.provide());
+       // read again
+       readMetadataItem(provider.provide());
```

Comment:
Please assert the same (cached) object is returned.

Reply:
Would it mean that I need to check whether the object returned from second call of provider.provider() is the same one from first call in order to confirm that this test method run through all codes of JsonFileMetadataItemsProvider.provider() ?

Reply:
Yes

Reply:
So sorry...